vmaas/rpm_pkg_names.go

```diff
@@ -0,0 +1,53 @@
+package vmaas
+
+import (
+	"slices"
+
+	"github.com/pkg/errors"
+	"github.com/redhatinsights/vmaas-lib/vmaas/utils"
+)
+
+type ContentData map[string][]string
+
+type RPMPkgNames struct {
+	Names      ContentData `json:"rpm_name_list"`
+	LastChange string      `json:"last_change"`
+}
+
+func (c *Cache) getContentData(names []string, contentSets []string) ContentData {
+	csMap := make(map[string]bool, len(contentSets))
+	for _, cs := range contentSets {
+		csMap[cs] = true
+	}
+
+	isDuplicate := make(map[string]bool, len(names))
+	contentData := make(ContentData, len(names))
+	for _, name := range names {
+		nameID, found := c.Packagename2ID[name]
+		if !found || isDuplicate[name] {
+			continue
+		}
+
+		csIDs := c.PkgNameID2ContentSetIDs[nameID]
+		labels := utils.ApplyMap(csIDs, c.ContentSetID2Label)
+		if len(contentSets) != 0 {
+			labels = utils.Intersection(labels, csMap)
+		}
+		slices.Sort(labels)
+		contentData[name] = labels
+		isDuplicate[name] = true
+	}
+	return contentData
+}
+
+func (req *PackageNamesRequest) rpmPkgNames(c *Cache) (*RPMPkgNames, error) { // TODO: implement opts
+	if len(req.RPMNames) == 0 {
+		return &RPMPkgNames{}, errors.Wrap(ErrProcessingInput, "'rpm_name_list' is a required property")
+	}
```

Comment:
it might not be correct, original implementation returns empty response when rpm_name_list is empty, isn't it?